TIL.md

```diff
@@ -0,0 +1,263 @@
+
+# Monday
+
+## Routing, template, components
+
+for picture, look in /TIL (ember.concepts.png)
+
+- router - is the best place to load data connected to the application path (/products - all products, /proudcts/1 - one product)
+- products router - loads data using model() hook (all products)
+- products template - we invoke component product-thumb for each element of products collection(available through model) with param product=product
+- product-thumb component - has access to product inside template.hbs and component.js (this.product)
+
+## Mirage
+- fake server that runs in client, useful in development, as with standard ember config of models and routes, it can be employed to simulate proper backend until it is implemented
+
+## Components
+- components can be generated using `ember g component --pod component-name`, pods are creating a neat structure of `tempate.hbs` and `component.js` in the `component-name` folder. Note: the name of a component has to contain a dash 
+
+# Thursday
+
+## Linters
+https://github.com/DockYard/eslint-plugin-ember-suave
+https://github.com/rwjblue/ember-cli-template-lint
+https://github.com/rwjblue/ember-template-lint
+- all in all - tools that are used to detect foul code style
+(cli-linter, eslinter, template-linter, suave)
+
+## Testing environment
+- qunit default was replaced with mocha (https://mochajs.org/ which is a test framework running on Node and in browser
+- chai, a assertion library, was also added
+- phantomjs, an old headless browser was replaced with the chrome, run in its headless mode
+(mocha, chrome headless, chai, phantomjs)
+
+## Runtime
+- node version manager - a way to manage multiple different node versions
+- yarn - package manager
+  - interesting feature of yarn interactive-update that allows selection of the packages to be updated
+(nvm, node, yarn, yarn interactive-update)
+
+# Friday
+
+## Continuous Integration and Deployment
+- travis was setup with proper config 
+  - repository were set
+  - `curl -o- -L https://yarnpkg.com/install.sh | bash` in `travis.yml` of ember app
+  - badges were added to respective repositories
+- both backend were deployed to heroku
+  - frontend
+    - automatic deployments if master is green
+    - ember build pack https://codon-buildpacks.s3.amazonaws.com/buildpacks/heroku/emberjs.tgz
+  - backend
+    - automatic deployments if master is green 
+    - standard configuration (database)
+
+## Javascript Transpilation
+https://guides.emberjs.com/v2.14.0/configuring-ember/build-targets/
+in `targets.js` it can be specified which browsers are supposed to work with app, 
+old code is bigger than new one, user of new browser do not necessarily want to have the option to run the app on older browsers. given which version is to be supported, the uglifier has to be updated accordingly 
+
+## Connecting with RoR backend
+* Backend
+- https://github.com/cyu/rack-cors
+- https://github.com/cerebris/jsonapi-resources
```

Comment:
Add link and checkbox (that you can check after reading the documentation), and fix links
[ ] [jsonapi resources](http://jsonapi-resources.com/v0.9/guide/)